com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Bob.java

```diff
@@ -763,262 +763,266 @@ else if (path.endsWith(".html")) {
+                } else {
+                    throw e;
                 }
-                errors.append(String.format("ERROR %s%s %s\n", taskResult.getTask().input(0),
-                        (taskResult.getLineNumber() != -1) ? String.format(":%d", taskResult.getLineNumber()) : "",
-                        message));
-                if (verbose) {
-                    if (taskResult.getException() != null) {
-                        errors.append("  ")
-                                .append(taskResult.getException().toString())
-                                .append("\n");
-                        StackTraceElement[] elements = taskResult
-                                .getException().getStackTrace();
-                        for (StackTraceElement element : elements) {
-                            errors.append("  ").append(element.toString())
+            }
+            for (TaskResult taskResult : result) {
+                if (!taskResult.isOk()) {
+                    ret = false;
+                    String message = taskResult.getMessage();
+                    if (message == null || message.isEmpty()) {
+                        if (taskResult.getException() != null) {
+                            message = taskResult.getException().getMessage();
+                        } else {
+                            message = "undefined";
+                        }
+                    }
+                    errors.append(String.format("ERROR %s%s %s\n", taskResult.getTask().input(0),
+                            (taskResult.getLineNumber() != -1) ? String.format(":%d", taskResult.getLineNumber()) : "",
+                            message));
+                    if (verbose) {
+                        if (taskResult.getException() != null) {
+                            errors.append("  ")
+                                    .append(taskResult.getException().toString())
                                     .append("\n");
+                            StackTraceElement[] elements = taskResult
+                                    .getException().getStackTrace();
+                            for (StackTraceElement element : elements) {
+                                errors.append("  ").append(element.toString())
+                                        .append("\n");
+                            }
                         }
                     }
                 }
             }
+            if (!ret) {
+                System.out.println("\nThe build failed for the following reasons:");
+                System.out.println(errors);
+            }
+            project.dispose();
+            return new InvocationResult(ret, result);
+        } finally {
+            TimeProfiler.createReport();
         }
-        if (!ret) {
-            System.out.println("\nThe build failed for the following reasons:");
-            System.out.println(errors);
-        }
-        project.dispose();
-        return new InvocationResult(ret, result);
```

Comment:
The code change here is very small, actually. I wrapped everything with the following:
```java
try {
    ...
} finally {
    TimeProfiler.createReport();
}
```